### ISSUE TITLE
Support a keep mask so padded / variable-length runs can be vmapped

### ISSUE THREAD

**Issue by AdamOrmondroyd** [AdamOrmondroyd/janesthetic#1]
## Support a `keep` mask, so runs of different length can be batched

To `vmap` over several runs they must share a shape, so shorter runs get padded — and a sampler that can decline to replace a point produces a variable number of dead points per step for the same reason. Either way the caller ends up with fixed-shape arrays plus a boolean mask, and needs `logZ` to ignore the padding.

Filtering with `x[keep]` is not an option: boolean indexing is dynamic-shape, so it cannot be jitted or vmapped, which is the whole point of this package.

### Masking the events is necessary but not sufficient

Zeroing the birth/death contributions in `compute_nlive` is the easy half:

```python
ev = jnp.concatenate([jnp.ones(n, int), -jnp.ones(n, int)])
ev = ev * jnp.concatenate([keep, keep]).astype(int)      # 0 = no event at all
```

The trap is `logdX`. With `t = 0` on masked rows, `logX` is flat along a masked run, so a real row followed by a masked one gets `logXm == logXp` and its shell collapses to `-inf`. Measured **-0.053 nats** against the filtered answer at 10% masking — small enough to look like noise, large enough to matter.

### The quadrature does work out

`logXp` is already correct: masked rows contribute nothing, so `logX[i-1]` is the volume after the previous *real* row. Only the look-ahead needs fixing — it must reach the next unmasked row, which is a reverse `cummin` over positions:

```python
pos = jnp.where(keep, jnp.arange(n), n)
nxt = jnp.concatenate([lax.cummin(pos[::-1])[::-1][1:], jnp.array([n])])
t_next = jnp.where(nxt < n, t[jnp.clip(nxt, 0, n - 1)], -jnp.inf)
logXm = logX + t_next          # flat along masked runs, so logX[i] + t[next real]
```

O(n), traceable, vmappable.

### Checked

Against dropping the same rows outright, on a 2000-point synthetic run:

| masked | filtered `logZ` | masked `logZ` | difference |
| --- | --- | --- | --- |
| 0% | -2.25317669 | -2.25317526 | 1.4e-06 |
| 10% | -2.25245094 | -2.25245094 | **0** |
| 35% | -2.23401308 | -2.23401308 | **0** |

Exact where it counts; the 1.4e-06 at 0% is float ordering, since that path reduces to the current one. `jit` and `vmap` both fine.

`keep=None` should keep today's behaviour untouched.
